app/src/main/java/com/example/ar_reshare/AddProduct.java

```diff
@@ -1,51 +1,109 @@
 package com.example.ar_reshare;
 
+import androidx.activity.result.ActivityResult;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.DialogFragment;
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 import androidx.recyclerview.widget.SortedList;
 import androidx.recyclerview.widget.SortedListAdapterCallback;
-
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
 import android.view.View;
 import android.widget.ArrayAdapter;
+import android.widget.Button;
 import android.widget.ImageView;
-
-import android.widget.LinearLayout;
 import android.widget.Spinner;
-
+import android.widget.Toast;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
-
-public class AddProduct extends AppCompatActivity {
-
-    private ArrayList<Integer> uploadedImages = new ArrayList<>();
+import java.util.Date;
+import java.util.List;
+
+public class AddProduct extends AppCompatActivity implements addPhotoDialog.NoticeDialogListener{
+
+    private ArrayList<Uri> uploadedImages = new ArrayList<>();
+    UploadImageAdapter adapter;
+    private final String CAMERA = "camera";
+    private final String GALLERY = "gallery";
+    ActivityResultLauncher<Intent> cameraActivityResultLauncher;
+    ActivityResultLauncher<String> galleryActivityResultLauncher;
+    File storageDir;
```

Comment:
Can you add visibility specifiers to some of these please? e.g. public vs private

Reply:
it's added now! thanks!